### DISCUSSION TITLE
Negative upper half-life estimates?

### DISCUSSION THREAD

**Discussion by aleighbrown** [erhard-lab/grandR#19] [Q&A] [answered]
First - wonderful that this exists, awesome, amazing, time saver. 

however, there are some weird results such as getting negative upper values for the half-life happening in my dataset?

```
small <- FilterGenes(exp1_grandR,use=1:100)             
small<-FitKinetics(exp1_grandR,
                         name = "with.loglik",
                         return.fields=c("Synthesis","Half-life","logLik"),
                         return.extra=function(d) c(
                             lower=d$conf.lower["Half-life"],
                             upper=d$conf.upper["Half-life"]
                         ))


```

This can be reproduced using this file  
[small_objects.RDS.txt](https://github.com/erhard-lab/grandR/files/10480882/small_objects.RDS.txt)
(I just added .txt to the end so it will upload to GitHub, change the extension and it's a normal RDS object) 

```
library(grandR)
library(data.table)
library(dplyr)

small = readRDS('small_objects.RDS')
test_me = GetAnalysisTable(small)

test_me |> as.data.table() |> melt(id.vars = c('Gene','Type','Length','Symbol')) |> 
    filter(value < 0 ) |> filter(grepl("Half-life",variable))

```

I haven't read through the whole pre-print so maybe this is expected?

**Comment by florianerhard** [comment, marked as answer]
Dear Anna-Leigh,
thanks for your interest. I checked, there are extremely noisy genes where the approximation used for CIs simply fails and reports negative estimates. We will replace this by 0 (for rates) or Infinity (for the Half-life) in the next release!
Best,
Florian

**Reply by aleighbrown** [reply]
Great - so I'll just label everything with a negative half-life as like "noisy" and ignore, thanks!
